### PR TITLE
Collect logs from spawned pods in CI.

### PR DESCRIPTION
**Pull request by markusthoemmes**
## Proposed Changes

To debug CI efficiently, the logs from the respective pods are incredibly helpful.

**Release Note**

```release-note
NONE
```

/assign @houshengbo

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Log got created here: https://console.cloud.google.com/storage/browser/_details/knative-prow/pr-logs/pull/knative-sandbox_operator/34/pull-knative-sandbox-operator-integration-tests/1254765990567219203/artifacts/k8s.log-e2e-tests.sh.txt

**Comment by houshengbo** [comment]
@markusthoemmes This looks awesome.
e2e-upgrade.sh has a function named `test-setup` as well. Can we make it call add_trap as well?
I am not sure if we can have functions with the same name, with one calling another.

**Comment by markusthoemmes** [comment]
@houshengbo looking at how Knative Serving uses this, that looks like a misuse of the test_setup function 🤔 . I'll look at it.

**Comment by houshengbo** [comment]
@markusthoemmes `test_setup` is located in the e2e-common.sh for serving. I only took some lines in it, and created test_setup in e2e-upgrade.sh of serving operator. Operator only needs these lines for upgrade tests, so I did not put them in e2e-common.sh for operator.

**Comment by markusthoemmes** [comment]
This is ready for another look @houshengbo

File: test/e2e-upgrade-tests.sh [hunk 1]
```diff
@@ -120,6 +120,19 @@ function test_setup() {
...
+  if ! which kail > /dev/null; then
+    bash <( curl -sfL https://raw.githubusercontent.com/boz/kail/master/godownloader.sh) -b "$GOPATH/bin"
+  fi
+
+  # Capture all logs.
+  kail > ${ARTIFACTS}/k8s.log-$(basename ${E2E_SCRIPT}).txt &
+  local kail_pid=$!
+  # Clean up kail so it doesn't interfere with job shutting down
+  add_trap "kill $kail_pid || true" EXIT
+
```
Comment (aliok) [inline test/e2e-upgrade-tests.sh:135]: Can we extract the contents of `test_setup` of `e2e-common.sh`  into a new function like `test_setup_logging` and use it in `test_setup` there and here?

Comment (markusthoemmes) [reply test/e2e-upgrade-tests.sh:135]: Sure, on it.

Comment (markusthoemmes) [reply test/e2e-upgrade-tests.sh]: Pushed. Let's wait and see if it still works.

**Comment by markusthoemmes** [comment]
Logs are collected!

see https://console.cloud.google.com/storage/browser/knative-prow/pr-logs/pull/knative-sandbox_operator/34/pull-knative-sandbox-operator-upgrade-tests/1255410258889674752/artifacts/ and https://console.cloud.google.com/storage/browser/knative-prow/pr-logs/pull/knative-sandbox_operator/34/pull-knative-sandbox-operator-integration-tests/1255401611040133121/artifacts/

**Comment by aliok** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**